logging/impl.go

```diff
@@ -32,6 +32,11 @@ type (
 		zapcore.Entry
 		fields []zapcore.Field
 	}
+
+	implWith struct {
+		logger    *impl
```

Comment:
Can you embed this field so we can leverage method-forwarding to avoid reimplementing all the methods as you have?

Reply:
done!